rasa/validator.py

```diff
@@ -192,6 +192,29 @@ def verify_utterances_in_stories(self, ignore_warnings: bool = True) -> bool:
 
         return everything_is_alright
 
+    def verify_actions_in_stories_rules(self) -> bool:
+        """Verifies that actions used in stories and rules are present in the domain."""
+        visited = set()
```

Comment:
why do we have to keep track of the visited items?

Reply:
true, it's not necessary - I kept it thinking this could process the for loop faster, I can remove this part.